### PR TITLE
Add http2.default_buffer_water_mark config to tune latency

### PR DESCRIPTION
**Pull request by randall**

### CODE REVIEW

**Comment by randall** [comment]
[approve ci fedora]

**Comment by randall** [comment]
@ezelkow1 Can you unstick this PR's CI?

**Comment by ezelkow1** [comment]
[approve ci fedora autest]

**Comment by ezelkow1** [comment]
@randall I dont think its stuck, just had another fedora failure. It looks like autest is running though so let that one finish first and then can retry just fedora

**Comment by masaori335** [comment]
@keesspoelstra is going to review this. Thanks in advance!

**Comment by masaori335** [comment]
[approve ci fedora]

**Comment by randall** [comment]
TODO: Add docs describing the tunable.

**Comment by randall** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
I will take a look at this PR too.

File: proxy/http2/Http2ConnectionState.cc [hunk 1]
```diff
@@ -1679,6 +1673,12 @@ Http2ConnectionState::send_a_data_frame(Http2Stream *stream, size_t &payload_len
     payload_length = 0;
   }
 
+  if (payload_length > 0 && this->session->is_write_high_water()) {
```
Comment (bryancall) [inline proxy/http2/Http2ConnectionState.cc:1676]: You are not going to send data to the client unless there is 32k to send?

File: doc/admin-guide/files/records.config.en.rst [hunk 2]
```diff
@@ -4269,6 +4269,12 @@ HTTP/2 Configuration
    frames. Write operation will be triggered at least once every this configured
    number of millisecond regardless of pending data size.
 
+.. ts:cv:: CONFIG proxy.config.http2.default_buffer_water_mark INT 32768
+   :reloadable:
+   :units: bytes
+
+   Specifies the buffer water mark size in bytes in the HTTP/2 layer.
```
Comment (bryancall) [inline doc/admin-guide/files/records.config.en.rst:4276]: I would like to see a better description.  One like: https://docs.trafficserver.apache.org/admin-guide/files/records.config.en.html#proxy-config-http-default-buffer-water-mark

File: proxy/http2/Http2ConnectionState.cc (hunk 1 above)
Comment (masaori335) [reply proxy/http2/Http2ConnectionState.cc:1676]: If you're wondering about the timing of the `this->session->flush()`, it is called with several conditions. 
- e.g. if the response body is 10 bytes, it'll be sent to the client immediately.

This is checking the high watermark of the session buffer to stop it keep growing.

File: doc/admin-guide/files/records.config.en.rst (hunk 2 above)
Comment (randall) [reply doc/admin-guide/files/records.config.en.rst]: Updated the doc blurb for this. 

There will be more docs coming to describe the why of this (in a subsequent PR)

**Review by bryancall — APPROVED** [review]

**Pull request merged**